tests/msgheaders_test.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { MsgHeaders } from "../nats-base-client/headers.ts";
+import {
+  assert,
+  assertEquals,
+  assertThrows,
+} from "https://deno.land/std@0.61.0/testing/asserts.ts";
+import { NatsError } from "../src/mod.ts";
+
+Deno.test("msgheaders - basics", () => {
+  const h = new MsgHeaders();
+  assertEquals(h.size(), 0);
+  assert(!h.has("foo"));
+  h.append("foo", "bar");
+  h.append("foo", "bam");
+  h.append("foo-bar", "baz");
+
+  assertEquals(h.size(), 3);
+  h.set("bar-foo", "foo");
+  assertEquals(h.size(), 4);
+  h.delete("bar-foo");
+  assertEquals(h.size(), 3);
+
+  let header = MsgHeaders.canonicalMIMEHeaderKey("foo");
+  assertEquals("Foo", header);
+  assert(h.has("Foo"));
+  assert(h.has("foo"));
+  const foos = h.values(header);
+  assertEquals(2, foos.length);
+  assert(foos.indexOf("bar") !== -1);
+  assert(foos.indexOf("bam") !== -1);
+
```

Comment:
Just to verify I would test that indexOf("bat") does return -1.

Reply:
for some reason, this is not updating here:
```javascript
const foos = h.values(header);
  assertEquals(2, foos.length);
  assertArrayContains(foos, ["bar", "bam"]);
  assert(foos.indexOf("baz") === -1);
```

Reply:
No, I saw you added that.